Src/Newtonsoft.Json.UnityConverters/PartialConverter/ValuesArray.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Newtonsoft.Json.UnityConverters
+{
+    public readonly struct ValuesArray<TInner> : IReadOnlyList<TInner>
+    {
+        private readonly TInner[] _array;
+
+        public int Count => _array.Length;
+
+        [MaybeNull]
+        public TInner this[int index]
+        {
+            get => _array[index];
+            set => _array[index] = value;
+        }
+
+        public ValuesArray(TInner[] array)
+        {
+            _array = array;
+        }
+
+        public ValuesArray(int capacity)
+        {
+            _array = new TInner[capacity];
+        }
+
+        [return: MaybeNull]
+        public TInner Get(int index)
+        {
+            return _array[index];
+        }
+
+        [return: NotNullIfNotNull("fallback")]
+        public T GetAsTypeOrDefault<T>(int index, [AllowNull] T fallback = default)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity.Converters/pullRequest?prid=5420094)